contrib/babelfishpg_tsql/sql/sys_function_helpers.sql

```diff
@@ -9727,28 +9727,37 @@ $BODY$
 LANGUAGE plpgsql
 STABLE;
 
-CREATE OR REPLACE FUNCTION sys.babelfish_conv_helper_to_varbinary(IN arg anyelement,
+CREATE OR REPLACE FUNCTION sys.babelfish_conv_helper_to_varbinary(IN typmod INTEGER,
+                                                                  IN arg anyelement,
                                                                   IN try BOOL,
                                                                   IN p_style NUMERIC DEFAULT 0)
 RETURNS sys.varbinary
 AS
 $BODY$
+DECLARE result sys.varbinary;
 BEGIN
     IF try THEN
-        RETURN sys.babelfish_try_conv_to_varbinary(arg, p_style);
+        RETURN sys.babelfish_try_conv_to_varbinary(typmod, arg, p_style);
     ELSE
         IF pg_typeof(arg) IN ('text'::regtype, 'sys.ntext'::regtype, 'sys.nvarchar'::regtype, 'sys.bpchar'::regtype, 'sys.nchar'::regtype) THEN
             RETURN sys.babelfish_conv_string_to_varbinary(arg, p_style);
         ELSE
-            RETURN CAST(arg AS sys.varbinary);
+            IF typmod = -1 THEN
+                RETURN CAST(arg as sys.varbinary);
+            ELSE
+                -- Pass typmod-4 (Way typmod for sys.binary/varbinary is stored) to helper functions to use the typmod in initial conversion
+                EXECUTE format('SELECT CAST($1 as sys.varbinary(%s))', typmod - 4) INTO result USING arg;
```

Comment:
I assume this `-4` is `VARHDRSZ` right? If so then we should pass correct typmod from the `C` code to this function so that we don't have to do that calculation here.

Reply:
+1 to this, instead of calculating here lets pass the actual tymod from C code.

Reply:
Fixed